lcm/meson.build

```
@@ -6,6 +6,12 @@ if unix
   lcm_extra_deps = [thread_dep]
 endif
 
+conf_data = configuration_data()
+conf_data.set('LCM_C_NAMESPACE', 'lcm')
```

This will require the user modifying this `meson.build` in order to set `LCM_C_NAMESPACE` to something else. I think that's OK because linking multiple versions of LCM into one binary seems is a pretty niche use-case. But if you did want to make this configurable like for the cmake support you could add a line to `meson_options.txt`:

```
option('LCM_C_NAMESPACE', type : 'string', value : 'lcm', description : 'The namespace of C symbols')
```

And update this to be:

```suggestion
conf_data.set('LCM_C_NAMESPACE', get_option('LCM_C_NAMESPACE'))
```